File: sleepy-test/java-junit/AsyncHelperTests_ContinuationErrorTimeoutNotHitTest.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertSame;

import java.util.ArrayList;
import java.util.List;

public class AsyncHelperTests {

    @Test
    public void continuationErrorTimeoutNotHitTest() throws Exception {
        List<Exception> exceptions = new ArrayList<>();

        RunnableWithTimeout cont = AsyncHelpers.withTimeout(
            AsyncHelpers.preventMultipleCalls(exceptions::add),
            50
        );

        ApplicationException exception = new ApplicationException("Foo");

        cont.run(exception);
        Thread.sleep(100);

        assertEquals(1, exceptions.size());
        assertNotNull(exceptions.get(0));
        assertSame(exception, exceptions.get(0));

        cont.run(null);
        cont.run(new ApplicationException("Some exception"));
        cont.run(null);
        cont.run(new ApplicationException("Some exception"));

        assertEquals(1, exceptions.size());
        assertNotNull(exceptions.get(0));
    }
}

File: refactored/duplicate-assert/python-pytest/AsyncHelperTests_ContinuationErrorTimeoutNotHitTest.py
import time
import pytest

class TestAsyncHelper:
    def test_continuation_error_timeout_not_hit_1(self):
        exceptions = []
        cont = AsyncHelpers.WithTimeout(AsyncHelpers.PreventMultipleCalls(exceptions.append), 50)
        exception = ApplicationException("Foo")
        cont.apply(exception)
        time.sleep(0.1)
        assert len(exceptions) == 1
        assert exceptions[0] is not None
        assert exceptions[0] is exception

    def test_continuation_error_timeout_not_hit_2(self):
        exceptions = []
        cont = AsyncHelpers.WithTimeout(AsyncHelpers.PreventMultipleCalls(exceptions.append), 50)
        exception = ApplicationException("Foo")
        cont.apply(exception)
        time.sleep(0.1)
        cont.apply(None)
        cont.apply(ApplicationException("Some exception"))
        cont.apply(None)
        cont.apply(ApplicationException("Some exception"))
        assert len(exceptions) == 1
        assert exceptions[0] is not None
